dataactcore/utils/jobQueue.py

```diff
@@ -1,36 +1,90 @@
-		if 'http://' not in validatorUrl:
-			validatorUrl = ''.join(['http://', validatorUrl])
+        # Set up url to the job queue to establish connection
+        queueUrl = ''.join(
+            [CONFIG_JOB_QUEUE['broker_scheme'], '://', CONFIG_JOB_QUEUE['username'], ':', CONFIG_JOB_QUEUE['password'],
+             '@', job_queue_url, ':', str(CONFIG_JOB_QUEUE['port']), '//'])
 
-		# Set up url to the job queue to establish connection
-		queueUrl = ''.join([CONFIG_JOB_QUEUE['broker_scheme'], '://', CONFIG_JOB_QUEUE['username'], ':', CONFIG_JOB_QUEUE['password'], '@', CONFIG_JOB_QUEUE['url'], ':', str(CONFIG_JOB_QUEUE['port']), '//'])
+        # Create remote connection to the job queue
+        self.jobQueue = Celery('tasks', backend=backendUrl, broker=queueUrl)
 
-		# Create remote connection to the job queue
-		self.jobQueue = Celery('tasks', backend=backendUrl, broker=queueUrl)
+        @self.jobQueue.task(name='jobQueue.enqueue')
+        def enqueue(jobID):
+            # Don't need to worry about the response currently
+            url = ''.join([validatorUrl, '/validate/'])
+            params = {
+                'job_id': jobID
+            }
+            response = requests.post(url, params)
+            return response.json()
 
-		@self.jobQueue.task(name='jobQueue.enqueue')
-		def enqueue(jobID):
-			# Don't need to worry about the response currently
-		    url = ''.join([validatorUrl, '/validate/'])
-		    params = {
-		        'job_id': jobID
-		    }
-		    response = requests.post(url,params)
-		    return response.json()
+        @self.jobQueue.task(name='jobQueue.generate_d1')
+        def generate_d_file(api_url, file_name, user_id, d_file_id, interface_holder, timestamped_name, skip_gen=False):
+            job_manager = interface_holder().jobDb
+
+            try:
+                # if skip_gen:
+                #     s3_url = s3UrlHandler().getSignedUrl(path="public/", fileName=file_name, method="GET")
+                #     job_manager.setDFileUrl(d_file_id, s3_url)
+                #     job_manager.setDFileStatus(d_file_id, "finished")
+                # else:
+                if not skip_gen:
+                    xml_response = str(requests.get(api_url, verify=False).content)
+                    url_start_index = xml_response.find("<results>", 0) + 9
+                    file_url = xml_response[url_start_index:xml_response.find("</results>", url_start_index)]
+                else:
+                    file_url = s3UrlHandler().getSignedUrl(path="public/", fileName=file_name, method="GET")
+
+                bucket = CONFIG_BROKER['aws_bucket']
+                region = CONFIG_BROKER['aws_region']
+
+                aws_file_name = "".join([str(user_id), "/", timestamped_name])
+
+                with open(timestamped_name, "wb") as file:
+                    # get request
+                    response = requests.get(file_url)
+                    # write to file
+                    file.write(response.content)
```

Comment:
FYI, prod instances are m3.xlarge: 15GB RAM, if that helps.

Katie Rose
UX/UI Developer
Strategic Innovations Group
Booz | Allen | Hamilton
301-825-7974 (o)
201-919-5120 (c)
rose_katherine@bah.com<mailto:rose_katherine@bah.com>

From: jworcestBAH <notifications@github.com>
Reply-To: fedspendingtransparency/data-act-broker-backend <reply@reply.github.com>
Date: Wednesday, August 3, 2016 at 10:05 PM
To: fedspendingtransparency/data-act-broker-backend <data-act-broker-backend@noreply.github.com>
Cc: "Rose, Katherine [USA]" <Rose_Katherine@bah.com>, Comment <comment@noreply.github.com>
Subject: [External] Re: [fedspendingtransparency/data-act-broker-backend] Db 800 d file endpoints (#174)


In dataactcore/utils/jobQueue.py<https://github.com/fedspendingtransparency/data-act-broker-backend/pull/174#discussion_r73452723>:

> +                    xml_response = str(requests.get(api_url, verify=False).content)

> +                    url_start_index = xml_response.find("<results>", 0) + 9

> +                    file_url = xml_response[url_start_index:xml_response.find("</results>", url_start_index)]

> +                else:

> +                    file_url = s3UrlHandler().getSignedUrl(path="public/", fileName=file_name, method="GET")

> +

> +                bucket = CONFIG_BROKER['aws_bucket']

> +                region = CONFIG_BROKER['aws_region']

> +

> +                aws_file_name = "".join([str(user_id), "/", timestamped_name])

> +

> +                with open(timestamped_name, "wb") as file:

> +                    # get request

> +                    response = requests.get(file_url)

> +                    # write to file

> +                    file.write(response.content)

This does seem to require holding the entire file in memory. That may be unavoidable here, but we should consider checking how much memory the validator instances have and confirming that the D1/D2 files will never be larger than that.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/fedspendingtransparency/data-act-broker-backend/pull/174/files/76ff235aeec23e8f624c1e78f85e9a176fb60727#r73452723>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMls6CnTktKGed-00553kyjVWxvcy3_aks5qcUkAgaJpZM4JcPYi>.